identity_credential/src/validator/domain_linkage_validator.rs

```diff
@@ -0,0 +1,485 @@
+  ///  - Validation of the matched Domain Linkage Credential fails.
+  pub fn validate_linkage<DOC: ValidatorDocument>(
+    issuer: &DOC,
+    configuration: &DomainLinkageConfiguration,
+    domain: &Url,
+    validation_options: &CredentialValidationOptions,
+  ) -> DomainLinkageValidationResult {
+    let mut matched_credentials = configuration
+      .linked_dids()
+      .iter()
+      .filter(|credential| credential.issuer.url().as_str() == issuer.did_str());
+
+    match matched_credentials.next() {
+      None => Err(DomainLinkageValidationError {
+        cause: DomainLinkageValidationErrorCause::InvalidStructure,
+        source: None,
+      }),
+      Some(credential) => {
+        if matched_credentials.next().is_some() {
+          Err(DomainLinkageValidationError {
+            cause: DomainLinkageValidationErrorCause::InvalidStructure,
+            source: None,
+          })
+        } else {
+          Self::validate_credential(issuer, credential, domain.clone(), validation_options)
+        }
+      }
+    }
+  }
+
+  /// Validates a [Domain Linkage Credential](https://identity.foundation/.well-known/resources/did-configuration/#domain-linkage-credential).
+  ///
+  /// *`issuer`: issuer of the credential.
+  /// *`credential`: domain linkage Credential to be verified.
+  /// *`domain`: the domain hosting the credential.
+  pub fn validate_credential<T: Serialize, DOC: ValidatorDocument>(
+    issuer: &DOC,
+    credential: &Credential<T>,
+    domain: Url,
+    validation_options: &CredentialValidationOptions,
+  ) -> DomainLinkageValidationResult {
+    let issuer_did: CoreDID =
+      CoreDID::parse(credential.issuer.url().to_string()).map_err(|err| DomainLinkageValidationError {
+        cause: DomainLinkageValidationErrorCause::InvalidIssuer,
+        source: Some(Box::new(err)),
+      })?;
+
+    CredentialValidator::validate(credential, issuer, validation_options, FailFast::AllErrors).map_err(|err| {
+      DomainLinkageValidationError {
+        cause: DomainLinkageValidationErrorCause::CredentialValidationError,
+        source: Some(Box::new(err)),
+      }
+    })?;
+
+    if credential.id.is_some() {
+      return Err(DomainLinkageValidationError {
+        cause: DomainLinkageValidationErrorCause::ImpermissibleIdProperty,
+        source: None,
+      });
+    }
+
+    // Validate type.
+    match &credential.types {
+      OneOrMany::Many(types) => {
+        if !types
+          .iter()
+          .any(|type_| type_ == DomainLinkageConfiguration::domain_linkage_type())
+        {
+          return Err(DomainLinkageValidationError {
+            cause: DomainLinkageValidationErrorCause::InvalidTypeProperty,
+            source: None,
+          });
+        }
+      }
+      OneOrMany::One(_) => {
+        Err(DomainLinkageValidationError {
+          cause: DomainLinkageValidationErrorCause::InvalidTypeProperty,
+          source: None,
+        })?;
+      }
+    };
```

Comment:
Since `CredentialValidator::validate` already checks for the existence of the `VerifiableCredential` type, this can be simplified to the following:
```suggestion
   // Validate type.
    if !credential.types.iter().any(|type_| type_ == DomainLinkageConfiguration::domain_linkage_type()) {
      return Err(DomainLinkageValidationError {
        cause: DomainLinkageValidationErrorCause::InvalidTypeProperty,
        source: None,
      });
    }
```